From af3ee40805ac41bc5639cd7578f90bd665e9e592 Mon Sep 17 00:00:00 2001
From: Namjae Jeon <namjae.jeon@samsung.com>
Date: Sat, 8 Dec 2012 14:54:35 +0900
Subject: [PATCH] f2fs: remove redundant call to f2fs_put_page in delete entry

Since, we anyway need to put the page after deleting entry. So, there is no
need to make same call under different conditions.
Move out the f2fs_put_page from the two conditions and call at once.

Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Amit Sahrawat <a.sahrawat@samsung.com>
---
 fs/f2fs/dir.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/fs/f2fs/dir.c b/fs/f2fs/dir.c
index 2a20c50..fc02d8b 100644
--- a/fs/f2fs/dir.c
+++ b/fs/f2fs/dir.c
@@ -514,10 +514,9 @@ void f2fs_delete_entry(struct f2fs_dir_entry *dentry, struct page *page,
 		ClearPageUptodate(page);
 		dec_page_count(sbi, F2FS_DIRTY_DENTS);
 		inode_dec_dirty_dents(dir);
-		f2fs_put_page(page, 1);
-	} else {
-		f2fs_put_page(page, 1);
 	}
+	f2fs_put_page(page, 1);
+
 	mutex_unlock_op(sbi, DENTRY_OPS);
 }
 
-- 
1.8.2.1.422.gdcd8c09

